hybridcloud(apitoken): add async delete method on apitoken model

mdtro opened the pull request:
Continuation of https://github.com/getsentry/sentry/pull/86069.
Use the newly created services in the model to process deletions.

mdtro committed:
hybridcloud(apitoken): add async delete method on apitoken model

markstory reviewed (commented):
Looking good, we should have a test to ensure that when an api token is deleted, that either the region_replica operation is called via mock patching, or by asserting that the ApiTokenReplica is also removed.

mdtro committed:
hybridcloud(apitoken): add test for apitoken replication of deletes

mdtro committed:
hybridcloud(apitoken): mock test for call to apitoken replica service

mdtro commented:
> Looking good, we should have a test to ensure that when an api token is deleted, that either the region_replica operation is called via mock patching, or by asserting that the ApiTokenReplica is also removed.

Added two tests. :)
